src/services/auth/plugins/password/utils.ts

```diff
@@ -49,6 +29,16 @@ export async function verifyCurrentPassword(
   return true;
 }
 
+/**
+ * Compares a plain password with a hash.
+ * @param password Password to encrypt.
+ * @param hash Hash to be compared against.
+ * @returns A promise to be either resolved with the comparison result salt or rejected with an Error
```

Comment:
We can use JS Doc to annotate the function with exceptions. 

```typescript
/**
 * Compares a plain password with a hash.
 * @param password The plain password to compare
 * @param hash Hash to be compared against.
 * @returns A promise to be either resolved with the comparison result salt
 * @throws Error if the comparison fails
 */
```

It would be great to have an ESLint rule to throw an error when we omit try-catch clause. It should have an ESLint rule to force developers to catch errors commented from JSDoc.

The bad news is that [JSDoc is no longer maintained in ESLint.](https://eslint.org/blog/2018/11/jsdoc-end-of-life/)
The good news is that [the suggested plugin could solve our issue](https://github.com/gajus/eslint-plugin-jsdoc/blob/HEAD/docs/rules/require-throws.md#readme)

The limitation is that currently, the developper can call a function that can throw, outside a try catch clause.. (There is no nested function verification)
[A 1 year old issue asking an enhancement is still open and doesn't seems to be active.](https://github.com/gajus/eslint-plugin-jsdoc/issues/1138)

Reply:
I've did some tests on the function (`utils.test.ts`), and I didn't been able to make it throws an exception without typecast. The `compareSync` function of `bcrypt` doesn't mention a possible error that can be throw. I think it is just a conventional way to mention that it returns a Promise. I've modified the comments to not mention the error.